visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -729,6 +752,54 @@ private DiffingMethodTolerance getDiffingMethodTolerance(CheckOptions checkOptio
     return sensitivity != null ? sensitivity : this.diffingMethodTolerance;
   }
 
+  private WebElement getClipElement(CheckOptions checkOptions) {
+    if (checkOptions.getClipElement() != null) {
+      return checkOptions.getClipElement();
+    } else if (checkOptions.getClipSelector() != null) {
+      return this.driver.findElement(By.cssSelector(checkOptions.getClipSelector()));
+    }
+
+    return null;
+  }
+
+  private byte[] cropImage(
+      byte[] imageData, String imageFormat, int x, int y, int width, int height) {
+    // We can check that before loading the image
+    if (x < 0 || y < 0) {
+      throw new VisualApiException("Clipped element is not visible.");
+    }
+
+    BufferedImage image = loadImage(imageData);
+
+    int imageWidth = image.getWidth();
+    int imageHeight = image.getHeight();
+    if (x + width > imageWidth || y + height > imageHeight) {
```

Comment:
I think we can still crop even if it's not fully on the screen (this will be better covered with FPS), but we can just `Math.min(imageWidth, x + width)`. I think this is what we're doing in other places.

Reply:
I updated the code to create a crop using the intersection of the visible viewport and clipped region.